internal/nodejs-action/generate-layout.sh

```diff
@@ -2,21 +2,35 @@
 
 set -euo pipefail
 
-package_scope=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f1)
-package_name=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f2)
-if [ "${package_name}" == "" ]; then
-	package_name="${package_scope}"
-	package_scope=""
+# The subject name is an npm package url (purl).
+# https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst
+#
+# The npm package's scope is considered a purl "namespace" and not part of the
+# package name. So the subject will take the form of:
+#
+# With scope:
+#   pkg:npm/<scope>/<name>@<version>
+#
+# Without scope:
+#   pkg:npm/<name>@<version>
+raw_package_scope=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f1)
+raw_package_name=$(echo "${PACK_JSON}" | jq -r '.[0].name' | cut -d'/' -f2)
+if [ "${raw_package_name}" == "" ]; then
```

Comment:
Given the examples above, it seems `raw_package_name` would never be empty?
`pkg:npm/<scope>/<name>@<version>` -> `<scope>`
`pkg:npm/<name>@<version>` -> `<name>@<version>`

Reply:
The name from the output of `npm pack` only contains the `<scope>/<package name>`. It's not a purl. The comment above was intended as a general comment about why I'm parsing out the parts of the name, and url encoding it etc.

Maybe I can add more comments to make that more clear.